Library/Homebrew/utils/sorbet.rb

```diff
@@ -1,8 +1,9 @@
 # typed: true
 # frozen_string_literal: true
 
-if ENV["HOMEBREW_SORBET_RUNTIME"]
-  Homebrew.install_bundler_gems!
+if ENV["HOMEBREW_SORBET_RUNTIME"] && ENV["HOMEBREW_BREW_FILE"]
+  # This is only supported under the brew environment.
```

Comment:
When else is this run?

Reply:
If you run rubocop outside of `brew`, it might have invoked gem installation if you happen to have `HOMEBREW_SORBET_RUNTIME` in your environment (which is a reasonable scenario to happen).

This limits it to only run under the brew environment - `HOMEBREW_BREW_FILE` should never be set by the user (we depend on that in other places too).

Reply:
`utils/sorbet` is a require in `rubocops.rb`.

Reply:
Gotcha. I kinda wonder if we should have separate directories for:
- stuff that we expect to be run when `HOMEBREW_BREW_FILE` is unset (e.g. used by `rubocops.rb`): maybe `standalone/`?
- stuff like what you've made above that's needed at `brew.rb` boot before e.g. `global.rb` is required: maybe `startup/`?
 
I think that'd make some of these implicit dependencies a bit more obvious. Thoughts?

Reply:
Currently the scenario is:

- Standalone: `load_path` and `utils/sorbet`
- Startup: `load_path`, `startup.rb` (which loads `config.rb` and Bootsnap) and `utils/sorbet`.

Splitting `utils/sorbet` is probably a good idea - it's just a bit annoying that `require "sorbet-runtime-stub"` needs hacks to avoid `sorbet-runtime` (but maybe we shouldn't be caring in the standalone case? Just let it happen?).

Reply:
> * Standalone: `load_path` and `utils/sorbet`
> * Startup: `load_path`, `startup.rb` (which loads `config.rb` and Bootsnap) and `utils/sorbet`.

I guess I'd propose:
- `standalone.rb`, `standalone/load_path`, `standalone/sorbet`
- `startup.rb`, `startup/config`, `startup/bootsnap`

?

> Splitting `utils/sorbet` is probably a good idea - it's just a bit annoying that `require "sorbet-runtime-stub"` needs hacks to avoid `sorbet-runtime` (but maybe we shouldn't be caring in the standalone case? Just let it happen?).

Either sounds good to me.